src/main/java/ch/epfl/biop/imaris/EasyXT.java

```diff
@@ -1287,88 +1284,62 @@ public static ISurfacesPrx create(ImagePlus imp, int timepointOffset) throws Err
-                int[] labels = LabelImages.findAllLabels(tImpLabel);
-                int[] voxelCounts = LabelImages.voxelCount(tImpLabel.getStack(), labels);
-
-                for (int idx = 0; idx < labels.length; idx++) {
-                    //System.out.println("************");
-                    //System.out.println(labels[idx]);
-                    //System.out.println(voxelCounts[idx]);
-                    if (voxelCounts[idx] > 3) {
-
-                        // duplicate and threshold a Label
-                        ImagePlus tempImage = tImpLabel.duplicate();
-                        IJ.setThreshold(tempImage, labels[idx], labels[idx]);
-                        IJ.run(tempImage, "Convert to Mask", "method=Default background=Dark black");// can't leave options blank, GUI pops-up
-
-                        // imaris requires binary 0-1
-                        int nProcessor = tempImage.getStack().getSize();
-                        IntStream.range(0, nProcessor).parallel().forEach(index -> {
-                            tempImage.getStack().getProcessor(index + 1).multiply(1.0 / 255.0);
-                        });
-
-                        // we don't need to check anymore if the binary
-                        // - contains pixel , thanks to LabelImages.findAllLabels()
-                        // - has > 1 voxel thanks to LabelImages.voxelCount() and the  if (voxelCounts[idx] > 1 )
-                        IDataSetPrx data = EasyXT.Dataset.create(tempImage);
-                        surface.AddSurface(data, t + timepointOffset);
-                        tempImage.close();
-                        // TODO: Warning: Because there is no way to set the Surfaces's IDs, there will certainly be a
-                        //  discrepancy between the id of an original surface and a modified surface returned using this method...
-                    } else if (voxelCounts[idx] == 1) {
-                        log.warning("Objects with a label " + labels[idx] + " has only 1 voxel and has been excluded (Imaris issue)");
-                    }
+            // Need the get the dataset to figure out the timestamps
+            IDataSetPrx currentDataset = Dataset.getCurrent();
+
+            for (int t = 0; t < timepoints; t++) {
+                // Make sure it doesn't try to get an invalid timepoint
+                if( currentDataset.GetSizeT() > t ) {
+                    labelImage.SetTimePoint( t, currentDataset.GetTimePoint( t ) );
                 }
-                tImpLabel.close();
             }
 
+            // Go through T and Z, produce 1D array of pixels to add to label image
+            for (int z = 0; z < slices; z++) {
+                for (int t = 0; t < timepoints; t++) {
+                    int idx = impLabel.getStackIndex(1, z + 1, t + 1);
+                    ImageProcessor ip = impLabel.getStack().getProcessor(idx);
+                    short[] pixels = (short[]) ip.getPixels();
```

Comment:
Short or unsigned shorts ?

Reply:
just shorts seems to do the trick from the current tests